src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/TargetFrameworkMoniker.cs

```diff
@@ -2,25 +2,200 @@
+        public static readonly TargetFrameworkMoniker Net50_Linux = Net50 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60 = Net50 with { FrameworkVersion = new Version(6, 0) };
+        public static readonly TargetFrameworkMoniker Net60_Linux = Net60 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60_Windows = Net60 with { Platform = Platforms.Windows };
+#pragma warning restore CA1707 // Identifiers should not contain underscores
 
-        private const string NetPrefix = "net";
+#pragma warning disable CA1034 // Nested types should not be visible
+        public static class Platforms
+        {
+            public const string Windows = "windows";
+            public const string Linux = "linux";
+        }
+#pragma warning restore CA1034 // Nested types should not be visible
 
-        public override string ToString() => Name;
+        private readonly string? _platform;
+        private readonly Version? _platformVersion;
+        private readonly string _framework;
+        private readonly Version _frameworkVersion;
 
-        public bool IsFramework => !IsNetStandard && !IsNetCore;
+        public TargetFrameworkMoniker(string framework, Version frameworkVersion)
+        {
+            _framework = framework;
+            _frameworkVersion = frameworkVersion;
+        }
 
-        public bool IsNetStandard => Name.StartsWith(NetStandardNamePrefix, StringComparison.OrdinalIgnoreCase);
+        public string Framework
+        {
+            get
+            {
+                if (IsFramework || IsNet50OrAbove)
+                {
+                    return Net;
+                }
+                else if (IsNetCoreApp)
+                {
+                    return NetCoreApp;
+                }
+                else if (IsNetStandard)
+                {
+                    return NetStandard;
+                }
+                else
+                {
+                    return Framework;
+                }
+            }
 
-        public bool IsNetCore => Name.StartsWith(NetPrefix, StringComparison.OrdinalIgnoreCase) && Name.Contains(".");
+            init => _framework = value;
+        }
 
-        public bool IsWindows => Name.Contains("windows");
+        public Version FrameworkVersion
+        {
+            get => NormalizeVersion(_frameworkVersion) ?? _frameworkVersion;
+            init => _frameworkVersion = value;
+        }
+
+        public string? Platform
+        {
+            get
+            {
+                if (IsWindows)
+                {
+                    return Platforms.Windows;
+                }
+
+                if (string.IsNullOrEmpty(_platform))
+                {
+                    return null;
+                }
+
+                return _platform;
+            }
+
+            init => _platform = value;
```

Comment:
guard clauses for valid platforms

Reply:
I'm not sure if we need that - it also keeps changing and I don't want to have to track that unless we *need* to know a valid option